spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -311,6 +359,354 @@ def _measure_axial_damage_ratio(self, im_data, p_lst, idx):
+            axes[idx_row, idx_col].text(x_ventral + 3,
+                                        y_ventral,
+                                        f'Ventral bridge\n{np.round(ventral_bridge_width_mm, 2)} mm',
+                                        color='red', fontsize=12, ha='left', va='bottom')
+
+        # Load the spinal cord segmentation mask
+        # The orientation is assumed to be RPI (because we reoriented the image to RPI using orient2rpi())
+        im_sc = Image(self.fname_sc)
+        im_sc_data = im_sc.data
+
+        # Get the total number of lesions; this will represent the number of rows in the figure. For example, if we have
+        # 2 lesions, we will have two rows. One row per lesion.
+        num_of_lesions = len(label_lst)
+        # Get the minimum sagittal slice with lesion. For example, if a lesion cover slices 7,8,9, get 7
+        min_sag_slice = min([min(self.tissue_bridges_plotting_data[lesion_idx].keys()) for
+                             lesion_idx in self.tissue_bridges_plotting_data])[0]
+        # Get the maximum sagittal slice with lesion. For example, if a lesion cover slices 7,8,9, get 9
+        max_sag_slice = max([max(self.tissue_bridges_plotting_data[lesion_idx].keys()) for
+                             lesion_idx in self.tissue_bridges_plotting_data])[0]
+        # Get the number of slices containing the lesion. For example, if a lesion cover slices 7,8,9, get 3
+        num_of_sag_slices = max_sag_slice - min_sag_slice + 1
+
+        # Get the midsagittal slice of the spinal cord
+        # Note: as the midsagittal slice is the same for all lesions, we can pick the first lesion ([0]) to get it
+        mid_sagittal_sc_slice = self.measure_pd.loc[0, 'midsagittal_spinal_cord_slice']
+
+        #  Create a figure
+        #  The figure has one row per lesion and one column per sagittal slice containing the lesion
+        fig, axes = plt.subplots(num_of_lesions,
+                                 num_of_sag_slices,
+                                 figsize=(num_of_sag_slices*5, num_of_lesions*5))
+
+        # Force axes to be a 2-dimensional array (to avoid indexing issues if we have only a single lesion or a single
+        # sagittal slice)
+        axes = np.atleast_2d(axes)
+
+        # Loop across lesions
+        for idx_row, lesion_label in enumerate(label_lst):
+            # NOTE: As the 'label_lesion()' function has been called at the beginning of the script, im_lesion_data is
+            # now "labeled" meaning that different lesions have different values, e.g., 1, 2, 3
+            # As we are looping across lesions, we get the lesion mask for the current lesion label
+            im_label_data_cur = im_lesion_data == lesion_label
+            # Restrict the lesion mask to the spinal cord mask (from anatomical level, it does not make sense to have
+            # lesion outside the spinal cord mask)
+            im_label_data_cur = im_label_data_cur * im_sc_data
```

Comment:
> Is there any chance that the SC could be soft?

Good question!

Right now, SCIseg outputs binary SC and binary lesion. But theoretically, users can use the contrast-agnostic model to get SC and then it would be soft. To be ready for this case, I updated the code based on your suggestion in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4489/commits/9bc734a56e10e3552e5f974aa14bed2838fc907c.